### PR TITLE
chore(hybrid-cloud): Remove Integration model dependency in IntegrationRepositoryProvider.

### PR DESCRIPTION
**Pull request by dashed**
Depends on https://github.com/getsentry/sentry/pull/52676

### CODE REVIEW

**Review by corps — APPROVED** [review]

File: tests/sentry/integrations/bitbucket/test_repository.py [hunk 1]
```diff
@@ -182,7 +186,8 @@ def test_create_repository_data_no_installation_id(self):
 
     def test_create_repository_data_integration_does_not_exist(self):
         integration_id = self.integration.id
-        self.integration.delete()
+        with exempt_from_silo_limits():
```
Comment (markstory) [inline tests/sentry/integrations/bitbucket/test_repository.py:189]: Should this be assume_test_silo_mode()?

File: tests/sentry/integrations/bitbucket_server/test_repository.py [hunk 2]
```diff
@@ -20,34 +20,35 @@
...
-            status=IdentityStatus.VALID,
-            data={
-                "consumer_key": "sentry-test",
-                "private_key": EXAMPLE_PRIVATE_KEY,
-                "access_token": "access-token",
-                "access_token_secret": "access-token-secret",
-            },
-        )
-        integration.add_organization(self.organization, self.user, default_auth_id=identity.id)
+        with exempt_from_silo_limits():
```
Comment (markstory) [inline tests/sentry/integrations/bitbucket_server/test_repository.py:30]: Should this be assume_test_silo_mode()?

File: tests/sentry/integrations/bitbucket_server/test_webhook.py [hunk 3]
```diff
@@ -107,7 +107,8 @@ def test_missing_integration(self):
         )
 
     def test_simple(self):
-        self.integration.add_organization(self.organization, default_auth_id=self.identity.id)
+        with exempt_from_silo_limits():
```
Comment (markstory) [inline tests/sentry/integrations/bitbucket_server/test_webhook.py:110]: Should this be assume_test_silo_mode()?

File: tests/sentry/integrations/gitlab/test_repository.py [hunk 4]
```diff
@@ -9,14 +9,15 @@
...
+from sentry.testutils.silo import exempt_from_silo_limits, region_silo_test
 from sentry.utils import json
 
 
-@control_silo_test(stable=True)
+@region_silo_test(stable=True)
 class GitLabRepositoryProviderTest(IntegrationRepositoryTestCase):
     provider_name = "integrations:gitlab"
 
+    @exempt_from_silo_limits()
```
Comment (markstory) [inline tests/sentry/integrations/gitlab/test_repository.py:20]: Should this be assume_test_silo_mode()?

File: tests/sentry/integrations/gitlab/test_repository.py [hunk 5]
```diff
@@ -150,7 +150,8 @@ def test_create_repository_data_no_installation_id(self):
 
     def test_create_repository_data_integration_does_not_exist(self):
         integration_id = self.integration.id
-        self.integration.delete()
+        with exempt_from_silo_limits():
```
Comment (markstory) [inline tests/sentry/integrations/gitlab/test_repository.py:153]: Should this be assume_test_silo_mode()?

**Review by markstory — APPROVED** [review]

**Pull request merged**